### ISSUE TITLE
Can not use the module when organization policy constraints/gcp.restrictNonCmekServices is active

### ISSUE THREAD

**Issue by felix-lecuyer** [terraform-google-modules/terraform-google-kubernetes-engine#1566]
### TL;DR

When the organization policy constraints/gcp.restrictNonCmekServices apply on services compute.googleapis.com and container.googleapis.com, there is no configuration that enables the deployment of a kubernetes cluster using the module. 

### Expected behavior

Given:
- a project where the organization policy constraints/gcp.restrictNonCmekServices apply on services compute.googleapis.com and container.googleapis.com
-  a terraform configuration using the kubernetes engine module and providing a boot_disk_kms_key pointing to a CMEK for nodepools and autoscale configuration

the configuration should result in the deployment of a kubernetes cluster, which nodes have disk encrypted using the provided CMEK. 

### Observed behavior

When running terraform apply, the following error message is returned:
Error: googleapi: Error 400: Failed precondition: Constraint `constraints/gcp.restrictNonCmekServices` violated for `projects/{project_id}` attempting to create a resource without specifying a KMS CryptoKey., failedPrecondition 

### Terraform Configuration

```hcl
module "gcp-network" {
  source       = "terraform-google-modules/network/google"
  version      = "6.0.1"
  project_id   = var.project_id
  network_name = var.network

  subnets = [
    {
      subnet_name           = var.subnetwork
      subnet_ip             = "10.128.0.0/20"
      subnet_region         = var.region
      subnet_private_access = true
    },
  ]

  secondary_ranges = {
    (var.subnetwork) = [
      {
        range_name    = var.ip_range_pods_name
        ip_cidr_range = "10.129.0.0/20"
      },
      {
        range_name    = var.ip_range_services_name
        ip_cidr_range = "10.130.0.0/20"
      },
    ]
  }
}

module "gke" {
  source                   = "terraform-google-modules/kubernetes-engine/google"
  version                  = "25.0.0"
  project_id               = var.project_id
  database_encryption      = [{ "key_name" : var.key_name, "state" : "ENCRYPTED" }]
  name                     = "main"
  region                   = var.region
  network                  = module.gcp-network.network_name
  subnetwork               = module.gcp-network.subnets_names[0]
  ip_range_pods            = var.ip_range_pods_name
  ip_range_services        = var.ip_range_services_name
  remove_default_node_pool = true
  create_service_account   = false
  service_account          = var.service_account
  node_metadata            = "metadata"
  kubernetes_version       = "latest"
  release_channel          = "REGULAR"
  cluster_autoscaling      = {
    enabled                    = true
    min_cpu_cores              = 1
    max_cpu_cores              = 4
    min_memory_gb              = 2
    max_memory_gb              = 6
    gpu_resources              = []
    auto_repair                = true
    auto_upgrade               = true
    auto_provisioning_defaults = {
      boot_disk_kms_key = var.kms_key_path
    }
  }
  node_pools = [
    {
      name              = "node-pool"
      machine_type      = "e2-medium"
      min_count         = 1
      max_count         = 2
      auto_upgrade      = true
      disk_size_gb      = 30
      boot_disk_kms_key = var.kms_key_path
    }
  ]
}
```

### Terraform Version

```sh
Terraform v1.3.7
```

### Additional information

Resolving this bug would result in enabling the usage of this module in [sovereign controls](https://cloud.google.com/sovereign-controls-by-partners/docs/overview) environments.

The failure in the deployment seems to come from 2 origins: 
- the module does not take into account the boot_disk_kms_key argument for the default node_pool created for initialization of the cluster. Reference: https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/master/cluster.tf#L266
- the module does not take into account auto_provisioning_defaults for the cluster autoscaling configuration. Reference:
https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/master/variables.tf#L228 & https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/master/cluster.tf#L104

**Comment by markusvanlaak** [comment]
+1

**Comment by bharathkkb** [comment]
Thanks for opening the issue and investigating!  Happy to review a PR adding support for these.

**Comment by felix-lecuyer** [comment]
Hi, 
Can we re-open this issue, as there is an open [Pull Request ](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/pull/1577) related to it and the fix is still needed?
